docs/ipfs.md

```diff
@@ -95,7 +95,72 @@ hello
+ => => transferring context: 2B                                                                                                                   0.0s
+ => [internal] load metadata for localhost:5050/ipfs/bafkreicq4dg6nkef5ju422ptedcwfz6kcvpvvhuqeykfrwq5krazf3muze:latest                           0.1s
+ => [1/2] FROM localhost:5050/ipfs/bafkreicq4dg6nkef5ju422ptedcwfz6kcvpvvhuqeykfrwq5krazf3muze@sha256:28bfa1fc6d491d3bee91bab451cab29c747e72917e  3.8s
+ => => resolve localhost:5050/ipfs/bafkreicq4dg6nkef5ju422ptedcwfz6kcvpvvhuqeykfrwq5krazf3muze@sha256:28bfa1fc6d491d3bee91bab451cab29c747e72917e  0.0s
+ => => sha256:7b1a6ab2e44dbac178598dabe7cff59bd67233dba0b27e4fbd1f9d4b3c877a54 28.57MB / 28.57MB                                                  2.1s
+ => => extracting sha256:7b1a6ab2e44dbac178598dabe7cff59bd67233dba0b27e4fbd1f9d4b3c877a54                                                         1.7s
+ => [2/2] RUN echo hello > /hello                                                                                                                 0.6s
+ => exporting to oci image format                                                                                                                 0.6s
+ => => exporting layers                                                                                                                           0.1s
+ => => exporting manifest sha256:b96d490d134221ab121af91a42b13195dd8c5bf941012d7bfe07eabcf5259eda                                                 0.0s
+ => => exporting config sha256:bd706574eab19009585b98826b06e63cf6eacf8d7193504dae75caa760332ca2                                                   0.0s
+ => => sending tarball                                                                                                                            0.5s
+unpacking docker.io/library/hello:latest (sha256:b96d490d134221ab121af91a42b13195dd8c5bf941012d7bfe07eabcf5259eda)...done
+> nerdctl run --rm -it hello cat /hello
+hello
+```
+
+#### Details about `localhost:5050/ipfs/<CID>`
+
+As of now, BuildKit doesn't support `ipfs://` prefix so nerdctl achieves builds on IPFS by having a read-only local registry backed by IPFS.
+This registry converts registry API requests to IPFS operations.
+So IPFS-agnostic tools can pull images from IPFS via this registry.
+
+When you specify `--ipfs` option to `nerdctl bulid`, it automatically starts the registry backed by the IPFS repo of the current `$IPFS_PATH`.
+By default, nerdctl exposes the registry at `localhost:5050`.
+You can change the address and can manually restart the registry using `nerdctl ipfs registry up` and `nerdctl ipfs registry down`.
+
+The following example changes the registry API address to `localhost:5555` instead of `localhost:5050`.
+
+```
+nerdctl ipfs registry down
+nerdctl ipfs registry up --registry-address=localhost:5555
+```
```

Comment:
We should recommend this tool for Kubernetes users tool, until we can get kube-apiserver and CRI to support  IPFS image refs